### PR TITLE
hapi v17

### PR DESCRIPTION
**Pull request by guischdi**

### CODE REVIEW

File: example/server.js [hunk 1]
```diff
@@ -1,42 +1,26 @@
...
-
-server.start((err) => {
-  if (err) throw err;
-  console.log(`example server started @ ${server.info.uri}`);
-});
+server.register(plugins)
+  .catch(console.error)
+  .then(() => console.log('Bootstraped all model routes'))
+  .then(() => server.start())
+  .then(() => console.log(`example server started @ ${server.info.uri}`));
```
Comment (fiws) [inline example/server.js]: how about just one `then` here?

File: example/server.js [hunk 2]
```diff
@@ -1,42 +1,26 @@
...
-server.connection({ port: 8080 });
+const server = new hapi.server({ port: 8080 });
 
 const plugins = [
   {
-    register: require('../forest'),
+    plugin: require('../forest'),
     options: {
-      bootstrap: [ require('./models/cat-model'), require('./models/user-model') ]
+      bootstrap: [ require('./models/cat-model') ] //, require('./models/user-model') ]
```
Comment (fiws) [inline example/server.js]: why was this commented out?

File: forest.js [hunk 3]
```diff
@@ -6,12 +6,13 @@ const stubJoi = require('./lib/stub-joi');
 const call = new require('call');
 const router = new call.Router();
 
-module.exports = (server, opts, next) =>  {
+module.exports = {};
```
Comment (fiws) [inline forest.js]: is this necessary?

File: handlers/getAll.js [hunk 4]
```diff
@@ -29,14 +29,14 @@ module.exports = (route, options) => {
...
-      poly => options.transformResponse(poly, req, reply) : undefined;
+      poly => options.transformResponse(poly, req) : undefined;
 
     const readStream = query.where(filter).lean().cursor({
       transform,
     }).pipe(jsonStream.stringify());
     const stream2 = new stream.Readable().wrap(readStream);
 
-    reply(stream2).type('application/json');
+    return stream2; //.type('application/json');
```
Comment (fiws) [inline handlers/getAll.js]: so. the mime time is just lost here?

File: package.json [hunk 5]
```diff
@@ -33,7 +34,7 @@
   "license": "MIT",
   "dependencies": {
     "JSONStream": "^1.1.4",
-    "joi": "^10.0.2",
+    "joi": "^13.0.2",
```
Comment (fiws) [inline package.json]: joi 13 requires node 8. that's why the travis tests are failing. can you bump the node version in the `.travis.yaml` file

File: example/server.js (hunk 1 above)
Comment (guischdi) [reply example/server.js]: We can only save one `.then` here, as the last `console.log` should be executed after `server.start` has been resolved.
But I'll do so, if you want.

File: forest.js (hunk 3 above)
Comment (guischdi) [reply forest.js]: would you like to indent the whole register function content (nearly the whole file) one more "tab"?

File: example/server.js (hunk 1 above)
Comment (guischdi) [reply example/server.js]: I removed the first `console.log`, as this is information is not so important

File: forest.js (hunk 3 above)
Comment (guischdi) [reply forest.js]: Ok, "initializing" `module.exports` is not needed, as I didn't know. Removed this line.

**Pull request merged**